aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -467,6 +467,43 @@ TORCH_META_FUNC(triangular_solve)(const Tensor& self, const Tensor& A, bool uppe
   }
 }
 
+TORCH_META_FUNC(_linalg_solve)(const Tensor& A,
+                               const Tensor& B,
+                               bool left) {
+  // dtype
+  at::native::checkFloatingOrComplex(A, "linalg.solve");
+  TORCH_CHECK(A.scalar_type() == B.scalar_type(),
+              "linalg.solve: Expected A and B to have the same dtype, but found A of type ",
+              A.scalar_type(), " and B of type ", B.scalar_type(), " instead");
+
+  // NumPy compat: Two types of 'B' tensors are supported:
+  // - 1D tensor or batch of 1D tensors (vector case)
+  // - 2D tensor or batch of 2D tensors (matrix case)
+  const bool vector_case = at::native::linalg_solve_is_vector_rhs(A, B);
+  auto B_ = vector_case ? B.unsqueeze(-1) : B;
+
+  // matrix shapes
+  at::native::checkInputsSolver(A, B_, /*left=*/left, "linalg.solve");
+
+  // Check that B can be broadcasted to the shape of A
+  auto B_broad_shape = std::get<0>(at::native::_linalg_broadcast_batch_dims(B_, A));
+  // We disallow the broadcasting of B as a vector when left=False as, in that case, A.shape = (*, 1, 1)
+  TORCH_CHECK(left || !vector_case, "linalg.solve: Vector broadcasting of the left hand side is not supported for left=False. In this case linalg.solve is equivalent to B / A.squeeze(-1)");
```

Comment:
Why don't we just implement this division without the warning?

Reply:
My take was that if people hit this, it's very like it is is because of an oversight in their code.
Also, we can always add it later if people request it.

Reply:
It is a valid case. A 1x1 matrix should work.

Reply:
My point is that, if they really want to do that, they should use plain division as it will be not only faster but it'll be more obvious what you're trying to do.

Reply:
I still disagree. What if you analyze subsets of a Gaussian matrix, where you might have variables not correlated with each other. Does it not work for torch.solve?